### PR TITLE
Security: Use Header.Set and Header.Del for X-Grafana-User header

### PR DESCRIPTION
**Pull request by beardhatcode**
This ensures that the X-Grafana-User header can be trusted.

If the configuration enabled the setting of this header, the
server can now trust that X-Grafana-User is set/unset by Grafana.

Before this, an anonymous user could simply set the X-Grafana-User
header themselves (using the developer tool for example)

---

**What this PR does / why we need it**: 

This edit is needed to ensure that the X-Grafana-User can be trusted.

**Which issue(s) this PR fixes**:

- The one I was going to create for this issue which I came across while attempting to ensure only logged-in users could access all metrics. Non logged in users can simply spoof the header using eg. the DevTools to add the header themselves.

**Special notes for your reviewer**:

Thanks for reviewing this. There is some code duplication here, maybe this could be fixed more cleanly but I'm not familiar enough with the codebase to know where this would go.

### CODE REVIEW

**Review by marefr — APPROVED** [review]
LGTM

**Review by marefr — CHANGES_REQUESTED** [review]
Was too fast reviewing. Please have a look at the comment.

File: pkg/api/pluginproxy/pluginproxy.go [hunk 1]
```diff
@@ -79,10 +79,14 @@ func NewApiPluginProxy(ctx *models.ReqContext, proxyPath string, route *plugins.
 			return
 		}
 
-		req.Header.Add("X-Grafana-Context", string(ctxJSON))
+		req.Header.Set("X-Grafana-Context", string(ctxJSON))
 
-		if cfg.SendUserHeader && !ctx.SignedInUser.IsAnonymous {
-			req.Header.Add("X-Grafana-User", ctx.SignedInUser.Login)
+		if cfg.SendUserHeader {
```
[the remark this answers is not in the document]
Comment (marefr) [reply pkg/api/pluginproxy/pluginproxy.go]: I know that. But looking at this now feels like a potential security problem that someone could send the X-Grafana-User from client when `SendUserHeader` is disabled. So I still would like to see this change, but I can help out with that and/or fix it in another PR. Your call.

[the remark this answers is not in the document]
Comment (beardhatcode) [reply pkg/api/pluginproxy/pluginproxy.go]: You are right, good call.

Now I always delete the header first and if logged in and the setting is enabled, the header is set.

**Review by marefr — APPROVED** [review]
LGTM

**Comment by marefr** [comment]
Thank you for contributing to Grafana!

**Pull request merged**